ntoskrnl/mm/ARM3/section.c

```diff
@@ -1635,14 +1635,12 @@ MiGetFileObjectForSectionAddress(
     if (Vad->u.VadFlags.Spare != 0)
     {
         PMEMORY_AREA MemoryArea = (PMEMORY_AREA)Vad;
-        PROS_SECTION_OBJECT Section;
 
         /* Check if it's a section view (RosMm section) */
         if (MemoryArea->Type == MEMORY_AREA_SECTION_VIEW)
         {
             /* Get the section pointer to the SECTION_OBJECT */
-            Section = MemoryArea->SectionData.Section;
-            *FileObject = ((PMM_SECTION_SEGMENT)Section->Segment)->FileObject;
+            *FileObject = MemoryArea->SectionData.Segment->FileObject;
```

Comment:
`SectionData.Segment` and `SectionData.Section->Segment` are guaranteed to be the same?

Reply:
Yes.